py/selenium/webdriver/common/bidi/browsing_context.py

```diff
@@ -66,12 +67,23 @@ def from_json(cls, json: dict) -> "NavigationInfo":
         -------
             NavigationInfo: A new instance of NavigationInfo.
         """
-        return cls(
-            context=json.get("context"),
-            navigation=json.get("navigation"),
-            timestamp=json.get("timestamp"),
-            url=json.get("url"),
-        )
+        context = json.get("context")
+        if context is None or not isinstance(context, str):
+            raise ValueError("context is required and must be a string")
+
+        navigation = json.get("navigation")
+        if navigation is not None and not isinstance(navigation, str):
+            raise ValueError("navigation must be a string")
+
+        timestamp = json.get("timestamp")
+        if timestamp is None or not isinstance(timestamp, int):
+            raise ValueError("timestamp is required and must be an integer")
```

Comment:
```suggestion
        timestamp = json.get("timestamp")
        if timestamp is None or not isinstance(timestamp, int) or timestamp < 0:
            raise ValueError("timestamp is required and must be a non-negative integer")
```
`timestamp` is of js-uint type, change it in `DownloadWillBeginParams` also

Reply:
I have made changes at both instances